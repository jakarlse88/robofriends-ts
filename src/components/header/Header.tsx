import * as React from 'react';
import CounterButton from '../counterbutton/CounterButton';

export default class Header extends React.Component {
    shouldComponentUpdate(nextProps: any, nextState: any): boolean {
        return false;
    }

    render() {
        return (
            <React.Fragment>
                <h1 className='f1'>RoboFriends</h1>
                <CounterButton color={'red'} />
            </React.Fragment>
        )
    }
}