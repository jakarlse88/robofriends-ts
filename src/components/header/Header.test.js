import { shallow } from "enzyme";
import Header from './Header';
import React from 'react';

describe('Header component', () => {
    it('expects to render the Header component', () => {
        expect(shallow(<Header />)).toMatchSnapshot();
    })
});