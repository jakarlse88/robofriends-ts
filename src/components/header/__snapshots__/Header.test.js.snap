// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component expects to render the Header component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1
          className="f1"
>
          RoboFriends
</h1>,
        <CounterButton
          color="red"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "RoboFriends",
          "className": "f1",
        },
        "ref": null,
        "rendered": "RoboFriends",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "red",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1
            className="f1"
>
            RoboFriends
</h1>,
          <CounterButton
            color="red"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "RoboFriends",
            "className": "f1",
          },
          "ref": null,
          "rendered": "RoboFriends",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "red",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
